x/ccv/provider/keeper/relay.go

```diff
@@ -223,18 +249,26 @@ func (k Keeper) OnRecvSlashPacket(ctx sdk.Context, packet channeltypes.Packet, d
 		panic(fmt.Errorf("SlashPacket received on unknown channel %s", packet.DestinationChannel))
 	}
 
-	// apply slashing
-	if _, err := k.HandleSlashPacket(ctx, chainID, data); err != nil {
-		errAck := channeltypes.NewErrorAcknowledgement(err.Error())
-		return &errAck
-	}
-
-	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
-	return ack
+	// Queue a pending slash packet entry to the parent queue, which will be seen by the throttling logic
+	k.QueuePendingSlashPacketEntry(ctx, providertypes.NewSlashPacketEntry(
+		ctx.BlockTime(), // recv time
+		chainID,         // consumer chain id that sent the packet
+		data.Validator.Address))
+
+	// Queue slash packet data in the same (consumer chain specific) queue as vsc matured packet data,
+	// to enforce order of handling between the two packet types.
+	k.QueuePendingSlashPacketData(ctx,
+		chainID,         // consumer chain id that sent the packet
+		packet.Sequence, // IBC sequence number of the packet
+		data)
```

Comment:
I agree, the naming isn't ideal.

It's a tradeoff because I believe the `Entry` / `Data` terms to be most important from an implementation perspective. Whereas the "global" vs "per chain" terminology is most useful from explaining the protocol from a high  level. 

If you look at what's being queued, an "entry" doesn't have any of the data to actually handle the slash packet, whereas the data queue does.

I could get behind `QueueGlobalSlashEntry`, `QueueChainSpecificSlashData`, and `QueueChainSpecificVSCMaturedData` giving us the benefit of all terms mentioned. What do you think?

Reply:
Sounds good! I would maybe just drop Entry and Data from the ends. The reason being that all queues have 'entries' and almost everything in software is 'data'.

Reply:
Good points, imo the importance of including "entry" and "data" in these methods is the concrete types they refer to: `SlashPacketData` (introduced before this PR), and `SlashPacketEntry` (new type). Maybe there's alternatives to how those types are named tho

Reply:
addressed in refactors